src/artemis/devices/unit_tests/test_grid_overlay.py

```diff
@@ -0,0 +1,149 @@
+    snapshot_filename = "snapshot"
+    snapshot_directory = "."
+    RE = RunEngine()
+    RE(take_snapshot_with_grid(oav, snapshot_filename, snapshot_directory))
+
+
+@pytest.mark.parametrize(
+    "top_left_x, top_left_y, box_width, num_boxes_x, num_boxes_y, expected_lines",
+    [
+        (
+            5,
+            5,
+            5,
+            5,
+            5,
+            [
+                ((5, 5), (5, 30)),
+                ((5, 5), (30, 5)),
+                ((5, 30), (30, 30)),
+                ((30, 5), (30, 30)),
+            ],
+        ),
+        (
+            10,
+            10,
+            10,
+            10,
+            10,
+            [
+                ((10, 10), (10, 110)),
+                ((10, 10), (110, 10)),
+                ((10, 110), (110, 110)),
+                ((110, 10), (110, 110)),
+            ],
+        ),
+    ],
+)
+@patch("src.artemis.devices.oav.grid_overlay.ImageDraw.Draw")
+def test_add_grid_border_overlay_to_image_makes_correct_calls_to_imagedraw(
+    mock_imagedraw: MagicMock,
+    top_left_x,
+    top_left_y,
+    box_width,
+    num_boxes_x,
+    num_boxes_y,
+    expected_lines,
+):
+    image = MagicMock()
+    add_grid_border_overlay_to_image(
+        image, top_left_x, top_left_y, box_width, num_boxes_x, num_boxes_y
+    )
+    expected_calls = 2 * [call(image)] + [
+        call(image).line(line) for line in expected_lines
+    ]
+    actual_calls = mock_imagedraw.mock_calls
+    expected_calls.sort()
+    actual_calls.sort()
+    assert expected_calls == actual_calls
+
+
+@pytest.mark.parametrize(
+    "top_left_x, top_left_y, box_width, num_boxes_x, num_boxes_y, expected_lines",
+    [
+        (
+            3,
+            3,
+            3,
+            3,
+            3,
+            [
+                ((3, 6), (12, 6)),
+                ((6, 3), (6, 12)),
+                ((3, 9), (12, 9)),
+                ((9, 3), (9, 12)),
+            ],
+        ),
+        (
+            4,
+            4,
+            4,
+            4,
+            4,
+            [
+                ((4, 8), (20, 8)),
+                ((4, 12), (20, 12)),
+                ((4, 16), (20, 16)),
+                ((8, 4), (8, 20)),
+                ((12, 4), (12, 20)),
+                ((16, 4), (16, 20)),
+            ],
+        ),
+    ],
+)
+@patch("src.artemis.devices.oav.grid_overlay.ImageDraw.Draw")
+def test_add_grid_overlay_to_image_makes_correct_calls_to_imagedraw(
+    mock_imagedraw: MagicMock,
+    top_left_x,
+    top_left_y,
+    box_width,
+    num_boxes_x,
+    num_boxes_y,
+    expected_lines,
+):
+    image = MagicMock()
+    add_grid_overlay_to_image(
+        image, top_left_x, top_left_y, box_width, num_boxes_x, num_boxes_y
+    )
+    expected_calls = 2 * [call(image)] + [
+        call(image).line(line) for line in expected_lines
+    ]
+    actual_calls = mock_imagedraw.mock_calls
+    expected_calls.sort()
+    actual_calls.sort()
+    assert actual_calls == expected_calls
```

Comment:
**nit**: Could you move the code shared between this and the above into a function they both call?